src/shogun/machine/RRForest.cpp

```diff
@@ -0,0 +1,177 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Written (W) 2016 Saurabh Mahindre
+ */
+
+#include <shogun/machine/RRForest.h>
+#include <shogun/multiclass/tree/RRCARTree.h>
+
+using namespace shogun;
+
+CRRForest::CRRForest()
+: CBaggingMachine()
+{
+	init();
+}
+
+CRRForest::CRRForest(int32_t rand_numfeats, int32_t num_bags)
+: CBaggingMachine()
+{
+	init();
+	
+	set_num_bags(num_bags);
+
+	if (rand_numfeats>0)
+		dynamic_cast<CRRCARTree*>(m_machine)->set_feature_subset_size(rand_numfeats);
+}
+
+CRRForest::CRRForest(CFeatures* features, CLabels* labels, int32_t num_bags, int32_t rand_numfeats)
+: CBaggingMachine()
+{
+	init();
+
+	SG_REF(features);
+	m_features=features;
+	set_labels(labels);
+
+	set_num_bags(num_bags);
+
+	if (rand_numfeats>0)
+		dynamic_cast<CRRCARTree*>(m_machine)->set_feature_subset_size(rand_numfeats);
+}
+
+CRRForest::CRRForest(CFeatures* features, CLabels* labels, SGVector<float64_t> weights, int32_t num_bags, int32_t rand_numfeats)
+: CBaggingMachine()
+{
+	init();
+
+	SG_REF(features);
+	m_features=features;
+	set_labels(labels);
+	m_weights=weights;
+
+	set_num_bags(num_bags);
+
+	if (rand_numfeats>0)
+		dynamic_cast<CRRCARTree*>(m_machine)->set_feature_subset_size(rand_numfeats);
+}
+
+CRRForest::~CRRForest()
+{
+}
+
+void CRRForest::set_machine(CMachine* machine)
+{
+	SG_ERROR("Machine is set as CRRCART and cannot be changed\n")
+}
+
+void CRRForest::set_weights(SGVector<float64_t> weights)
+{
+	m_weights=weights;
+}
+
+SGVector<float64_t> CRRForest::get_weights() const
+{
+	return m_weights;
+}
+
+EProblemType CRRForest::get_machine_problem_type() const
+{
+	REQUIRE(m_machine,"m_machine is NULL. It is expected to be RRCARTree\n")
+	return dynamic_cast<CRRCARTree*>(m_machine)->get_machine_problem_type();
+}
+
+void CRRForest::set_machine_problem_type(EProblemType mode)
+{
+	REQUIRE(m_machine,"m_machine is NULL. It is expected to be RRCARTree\n")
+	dynamic_cast<CRRCARTree*>(m_machine)->set_machine_problem_type(mode);
+}
+
+void CRRForest::set_num_random_features(int32_t rand_featsize)
+{
+	REQUIRE(m_machine,"m_machine is NULL. It is expected to be RRCARTree\n")
+	REQUIRE(rand_featsize>0,"feature subset size should be greater than 0\n")
```

Comment:
Please be more careful with error messages.
`REQUIRE(rand_featsize>0, "Feature subset size (%d) should be greater than 0.\n", rand_featsize)`